lib/yard-gobject-introspection.rb

```diff
@@ -5,8 +5,12 @@
 class GObjectIntropsectionHandler < YARD::Handlers::Ruby::Base
   handles :module
 
+  @@has_processed_module = false
```

Comment:
This doesn't work when the target library loads multiple `.gir` files.
https://github.com/ruby-gnome/ruby-gnome/tree/master/pango is an example of this case.

Reply:
Ok, I changed it to keep a list of modules instead. This will make the Arrow module be processed for the `red-parquet`. Is there a way to know what modules are actually loaded?

Reply:
We can detect it by checking arguments of `GObjectIntrospection::Loader.load`.
For example: https://github.com/apache/arrow/blob/bcaeaa8c2d970b81249cfba019475598e3d3109f/ruby/red-parquet/lib/parquet/loader.rb#L22

But it's dynamic and it may be in a method body like `red-parquet` did. So detecting it automatically may be hard.
We may need to write something as a comment for `yard-gobject-introspection`...

Anyway, this approach is enough for now.